sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4957,6 +4957,9 @@ enqueueMemFillHelper(pi_command_type CommandType, pi_queue Queue, void *Ptr,
   ze_command_list_handle_t ZeCommandList = nullptr;
   ze_fence_handle_t ZeFence = nullptr;
 
+  // TODO: Fill operations on copy engine fails to fill a buffer at expected
+  // offset. Perform fill operations on compute engine for now.
+  // PreferCopyEngine will be initialized with 'true' once issue is resolved.
```

Comment:
If we turn it to true here then the check for max pattern size needs to be able to set it false. So, probably, invert the condition there.

Reply:
That's a good point. Done.